SourceX/controls/devices/kbcontroller.cpp

```diff
@@ -38,22 +38,26 @@ ControllerButton KbCtrlToControllerButton(const SDL_Event &event)
 #endif
 #ifdef KBCTRL_BUTTON_A
 		case KBCTRL_BUTTON_A:
+			if (IsModifierKey())
+				return ControllerButton::BUTTON_QSPELL3;
 			return ControllerButton::BUTTON_A;
 #endif
 #ifdef KBCTRL_BUTTON_B
 		case KBCTRL_BUTTON_B: // Right button
 			if (IsModifierKey())
-				return ControllerButton::BUTTON_RIGHTSTICK;
+				return ControllerButton::BUTTON_QSPELL4;
 			return ControllerButton::BUTTON_B;
 #endif
 #ifdef KBCTRL_BUTTON_X
 		case KBCTRL_BUTTON_X: // Left button
 			if (IsModifierKey())
-				return ControllerButton::BUTTON_BACK;
+				return ControllerButton::BUTTON_QSPELL1;

```

Comment:
Looks like this breaks quest log and virtual mouse left click on RG300

Reply:
I haven't used either for the past few builds and I think this is a better trade off. The only use I see for mouse emulation at this point is for long teleportation and telekinesis, I could improve on them but I think it would be overkill.